Remove .gitignore file from dist package

cedric-anne opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Not a big deal, but `.gitignore` file is present in dist package fetched from packagist.

cedric-anne committed:
Remove .gitignore file from dist package

cedric-anne commented:
Sorry for the multiple commits, I have made it one by one from GitHub UI. I can squash them if it is a problem.

GromNaN reviewed (approved):
> Sorry for the multiple commits, I have made it one by one from GitHub UI. I can squash them if it is a problem.

Don't worry, we have a tool to squash before merging.

nicolas-grekas commented:
There are more of them:
```
src/Symfony/Contracts/Cache/.gitignore
src/Symfony/Contracts/.gitignore
src/Symfony/Contracts/EventDispatcher/.gitignore
src/Symfony/Contracts/Service/.gitignore
src/Symfony/Contracts/Translation/.gitignore
src/Symfony/Contracts/Deprecation/.gitignore
src/Symfony/Contracts/HttpClient/.gitignore
```

cedric-anne commented:
I added `.gitattributes` files where they were missing, and also added `export-ignore` on test files.
By the way, I noticed that some the namespace of test files is sometimes `Test` on `Symfony\Contracts`, and that is not consistent with namespace `Tests` used everywhere else in Symfony codebase.

cedric-anne commented:
> By the way, I noticed that some the namespace of test files is sometimes `Test` on `Symfony\Contracts`, and that is not consistent with namespace `Tests` used everywhere else in Symfony codebase.

My bad, these should not be excluded.

fabpot commented on src/Symfony/Contracts/Translation/.gitattributes:
```diff
@@ -0,0 +1,3 @@
+/Test export-ignore
```
Should be reverted. The `Test` namespace contains files that people can reuse in their own tests. The `Tests` namespace contains tests that we can indeed ignore.

nicolas-grekas commented:
Thank you @cedric-anne.
